ibis/backends/__init__.py

```diff
@@ -223,6 +224,43 @@ def to_pyarrow(
             table.rename_columns(table_expr.columns).cast(arrow_schema)
         )
 
+    @util.experimental
+    def to_polars(
+        self,
+        expr: ir.Expr,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        **kwargs: Any,
+    ) -> pl.DataFrame:
+        """Execute expression and return results in as a polars DataFrame.
+
+        This method is eager and will execute the associated expression
+        immediately.
+
+        Parameters
+        ----------
+        expr
+            Ibis expression to export to polars.
+        params
+            Mapping of scalar parameter expressions to value.
+        limit
+            An integer to effect a specific row limit. A value of `None` means
+            "no limit". The default is in `ibis/config.py`.
+        kwargs
+            Keyword arguments
+
+        Returns
+        -------
+        dataframe
+            A polars DataFrame holding the results of the executed expression.
+
+        """
+        import polars as pl
+
+        table = self.to_pyarrow(expr.as_table(), params=params, limit=limit, **kwargs)
```

Comment:
Is `expr.as_table()` required here?

Reply:
Ensuring the output is always a `pyarrow.Table` here makes massaging the results into the proper DataFrame/Series/Scalar values much more straightforward. It keeps that logic localized to the data mapper and not here.